root/meta/race-12552/exec_dlopenrace.C

```diff
@@ -0,0 +1,47 @@
+#include <thread>
+#include "TClassTable.h"
+#include "TROOT.h"
+#include <dlfcn.h>
+#include <iostream>
+
+void repeatopen(int n)
+{
+   for(int i = 0; i < n; ++i) {
+     auto r = dlopen("./libUser.so", RTLD_LAZY);
+     if (!r) {
+        std::cerr << "Failed to open libUser.so at iteration " << i << "\n";
+        return;
+     }
+     // Waste some time
+     for(int j = 0; j < 10; ++j)
+       auto d = TClassTable::GetDictNorm("UserClass");
+     // std::cerr << "Seeing dict pointer: " << (void*)d << '\n';
+     dlclose(r);
```

Comment:
This uses the platform-specific `dlopen()` and `dlclose()`.  Can't we make it platform-independent?  As @bellenot mentioned, in its current state, the test will not build on Windows.

Alternatively, the test should be disabled on non-POSIX platforms.

Reply:
It is now disable on WIndows.  I elect not to run on Windows as I can use the ROOT platform independent interface (they do too much) and running/testing on one platform is sufficient to test this race condition.